### ISSUE TITLE
Update Acerra, Italy to public source (from our S3 bucket)

### ISSUE THREAD

**Issue by smemorex** [openaddresses/openaddresses#2318]
hello,
@nvkelso create form me to a csv file now published on openaddresses and that will be imported ( now there is some problem) 

ho can i create this file on my own in case of modification and public it on municipality official site so openaddresses can take it from there automatically?

**Comment by nvkelso** [comment]
Once the source is published with a public URL, we'll update it in the OpenAddress source:

- https://github.com/openaddresses/openaddresses/blob/master/sources/it/04/acerra.json#L9

(Original work for this was done in https://github.com/openaddresses/openaddresses/issues/2257.)

**Comment by nvkelso** [comment]
Adding @smemorex comment from https://github.com/openaddresses/openaddresses/issues/2257#issuecomment-278741418 here instead:

hello,
can i upload zip file directly on official site? like this is ok

http://servizi.comune.acerra.na.it:8088/acerra.zip

?

if i need to add new number i think i can simply add new line but how can i
generate hash?


2016-12-22 0:42 GMT+01:00 Nathaniel V. KELSO <notifications@github.com>:

**Comment by nvkelso** [comment]
@smemorex Yes, a public URL like `http://servizi.comune.acerra.na.it:8088/acerra.zip` would work for us.

It might be useful to indicate what's in the zip file for humans. Like `accerra-address-points.zip`, but the machine (e.g.: https://github.com/openaddresses/openaddresses/blob/master/sources/it/04/acerra.json#L9) doesn't care what the filename is.

**Comment by smemorex** [comment]
hello , file renamed , can you tell me how to generate hash so i can update file?

**Comment by nvkelso** [comment]
As long as that is the final file name (maybe you want it in Italian?) I
think OpenAddresses just needs to update our source URL for it from S3 to
your official gov’t URL.

**Comment by smemorex** [comment]
hello,
we need to update the csv , i understand how to do all but how don't understand only how to create hash , can you tell me please so i can update with other street and housenumber? i see an hash column .

**Comment by nvkelso** [comment]
The CSV includes latitude and longitude coordinates, and the address text:

```
LON,LAT,NUMBER,STREET,UNIT,CITY,DISTRICT,REGION,POSTCODE,ID,HASH
14.3932886,40.954425,83,VIA PIETRABIANCA,,,,,,,ab08b5ba582553c6
14.3934445,40.9545203,85,VIA PIETRABIANCA,,,,,,,cd625b14c8e47867
14.3935504,40.9545784,87,VIA PIETRABIANCA,,,,,,,81c54f33f254e573
14.3892321,40.9522965,43,VIA PIETRABIANCA,,,,,,,3123972a8137f21b
```

It also includes a VRT that enables QGIS to open this data via GDAL/OGR. Here it is plotted on OSM background:

<img width="777" alt="screen shot 2017-02-10 at 09 09 46" src="https://cloud.githubusercontent.com/assets/853051/22836337/d0c46c10-ef70-11e6-834b-942955fa589b.png">

The existing file we have in S3 was a SHP format "shapefile", which is a more standard distribution format for geographic data than CSV + VRT. It's probably better to distribute a SHP file, and that conversion can be done using GDAL/ORG on the command line once installed.

@iandees What does Machine prefer, can it take the CSV + VRT or does it need a shapefile?

**Comment by iandees** [comment]
CSV by itself is fine.

**Comment by nvkelso** [comment]
@smemorex meaning to say the ZIP you've posted already works for us.

**Comment by nvkelso** [comment]
@smemorex Please confirm the final file name & URL and we'll update the OpenAddress configuration file.

**Comment by smemorex** [comment]
Ok, but i don't uderstand  how ti update It, in particolar how ti generate
hash column

Il 10/feb/2017 20:15, "Nathaniel V. KELSO" <notifications@github.com> ha
scritto:

**Comment by iandees** [comment]
We generate the hash column for you.

**Comment by smemorex** [comment]
hello, How Can i generate on my own?

Il 10/feb/2017 20:48, "Ian Dees" <notifications@github.com> ha scritto:

**Comment by iandees** [comment]
Check Google for `md5 hash` or `sha1 hash` to learn more about generating hashes.

**Comment by smemorex** [comment]
Ok i know It, i have to do md5 of What? A cumn valute or all'?

Il 10/feb/2017 08:56 PM, "Ian Dees" <notifications@github.com> ha scritto:

**Comment by iandees** [comment]
It depends on what you want to make the hash out of.

We're going to generate our own hash regardless of if the CSV we download has one, so there's no need for you to do it.

**Comment by smemorex** [comment]
Hello, if i have to add a point with street, housenumber, coordinate, i see
all on. CSV so i can add It but how Can i generate hash so i can load in my
pelias install and share with openaddresses?Is hashes needed?

Il 10/feb/2017 21:00, "Ian Dees" <notifications@github.com> ha scritto:

**Comment by nvkelso** [comment]
The data is collected by OpenAddresses, and then Pelias consumes OpenAddresses (not the other way around).

**Comment by migurski** [comment]
@smemorex, please don’t generate your own hash column. It’s internally calculated by OpenAddresses.

**Comment by nigelhorne** [comment]
Where is the algorithm for creating the hash documented?
